File: examples/examples_SSE_FORTRAN/gm29_sse.f90

PROGRAM GM29_Example

   PARAMETER (N=100000000)
   INTEGER i,GM29_SSE_generate,sum
   REAL val,rsum

   TYPE GM29_state
     INTEGER z(68)
   END TYPE GM29_state
   TYPE(GM29_state) state, SSE_state

   CALL GM29_init ( state )
   CALL GM29_Get_SSE_state (state,SSE_state)
   CALL GM29_Print_State (state)

   sum = 0

   DO i=1,N
     sum = sum + GM29_SSE_generate (SSE_state)
   END DO

   WRITE(*,1) N
   rsum = sum/4294967296.
   val = GM29_SSE_generate (SSE_state)/4294967296.
   IF (rsum<0) THEN
     rsum=rsum+1
   ENDIF
   IF (val<0) THEN
     val=val+1
   ENDIF
   WRITE(*,2) rsum
   WRITE(*,3) val

1  FORMAT(I9," GM29_SSE pseudorandom numbers generated.")
2  FORMAT("Fractional part of the total sum of generated numbers: ",F8.6)
3  FORMAT("Next output value: ",F8.6)

END PROGRAM GM29_Example

File: tpl/src/rngsse2/examples/examples_SSE_C/gm29_sse.c

#include<stdio.h>
#include<gm29.h>

#define NN       100000000UL

int main(void){ 
   long i; unsigned int sum=0;
   gm29_state state; gm29_sse_state sse_state;
   gm29_init_(&state); gm29_get_sse_state_(&state,&sse_state);
   gm29_print_state_(&state);
   for(i=0;i<NN;i++) sum+=gm29_sse_generate_(&sse_state);
   printf("%ld GM29_SSE pseudorandom numbers generated.\n",NN);
   printf("Fractional part of the total sum of generated numbers: %f\n",sum/4294967296.);
   printf("Next output value: %f\n",gm29_sse_generate_(&sse_state)/4294967296.);
   return 0;
}
